src/constitutive/TACSPhaseChangeMaterialConstitutive.cpp

```diff
@@ -0,0 +1,292 @@
+/*
+  This file is part of TACS: The Toolkit for the Analysis of Composite
+  Structures, a parallel finite-element code for structural and
+  multidisciplinary design optimization.
+
+  Copyright (C) 2010 University of Toronto
+  Copyright (C) 2012 University of Michigan
+  Copyright (C) 2014 Georgia Tech Research Corporation
+  Additional copyright (C) 2010 Graeme J. Kennedy and Joaquim
+  R.R.A. Martins All rights reserved.
+
+  TACS is licensed under the Apache License, Version 2.0 (the
+  "License"); you may not use this software except in compliance with
+  the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+#include "TACSPhaseChangeMaterialConstitutive.h"
+
+const char* TACSPhaseChangeMaterialConstitutive::psName =
+    "TACSPhaseChangeMaterialConstitutive";
+
+const char* TACSPhaseChangeMaterialConstitutive::getObjectName() {
+  return psName;
+}
+
+/*
+  PhaseChangeMaterialStiffness member function definitions
+*/
+TACSPhaseChangeMaterialConstitutive::TACSPhaseChangeMaterialConstitutive(
+    TACSMaterialProperties* solid_props, TACSMaterialProperties* liquid_props,
+    TacsScalar _lh, TacsScalar _Tm, TacsScalar _dT, TacsScalar _t, int _tNum,
+    TacsScalar _tlb, TacsScalar _tub) {
+  solid_properties = solid_props;
+  if (solid_properties) {
+    solid_properties->incref();
+  }
+  liquid_properties = liquid_props;
+  if (liquid_properties) {
+    liquid_properties->incref();
+  }
+  lh = _lh;
+  Tm = _Tm;
+  t = _t;
+  tNum = _tNum;
+  tlb = _tlb;
+  tub = _tub;
+  dT = _dT;
+  b = 2.0 * tan(0.99 * M_PI / 2.0);
```

Comment:
Can we add some comments to the code explaining how the formulation works? What the arctan is used for, each of the terms in the equation, etc.?